functionary/prompt.py

```diff
@@ -37,12 +37,27 @@ def from_message(cls, message: Dict) -> EndToken:
         elif role == "function":
             return EndToken.function
         else:  # role = assistant
-            if "function_call" in message:
+            if "function_call" in message and message["function_call"] is not None:
                 return EndToken.function_call
             else:
                 return EndToken.assistant
 
 
+class BeginToken(str, Enum):
+    function = "<|BEGIN_OF_FUNCTION_CALL|>"
```

Comment:
Lets also rename this to be start. and all related code too

Reply:
Sure